File: leetcode/medium/2125/2125.ts
export function numberOfBeams(bank: string[]): number {
  let beams: number = 0;
  let prev: number = 0;

  for (const floor of bank) {
    const curr = floor.replaceAll("0", "").length;
    beams += prev * curr;
    prev = curr || prev;
  }

  return beams;
}

const a = numberOfBeams(["011001", "000000", "010100", "001000"]);
console.log(a);

File: leetcode/medium/2125/2125.go
package medium2125

func numberOfBeams(bank []string) int {
    beams := 0
	prev := 0

	for _, floor := range bank {
		curr := 0
		for _, ch := range floor {
			if ch == '1' {
				curr++
			}
		}

		
		if curr > 0 {
			beams += prev * curr
			prev = curr
		}
	}

	return beams
}